### DISCUSSION TITLE
TODO 리스트

### DISCUSSION THREAD

**Discussion by YiBeomSeok** [CaZaIt/CaZaIt-Android-Private#153] [General]
# TODO 리스트
- 1. 우선순위: 버그 해결, 추가할 것
- 2. 리팩토링

## 발견된 문제
- [x] 카페 맵 프래그먼트에서 카페 인포 페이지로 넘어갔다가 다시 돌아왔을 때 마커들이 사라지는 문제
- [x] 토큰 재발급이 잘 안 되는 문제
- [ ] CafeList 페이지에서 리사이클러뷰 갱신이 제대로 안 되는 문제 (viewmodel의 liveData를 제대로 옵저브 하지 못 하는 것 같음)

## 추가해야 할 것
- [x] 햄버거 바(메뉴 버튼) 알림 확인 버튼으로 전환하기, 알림 확인 페이지 만들기
- [x] 필터 버튼 활성화하기
- [ ] 리뷰 수정 및 삭제
- [x] 계정관리 페이지
- [x] 아이디 찾기
- [x] 비밀번호 찾기

## 리펙토링할 것
- [x] 리뷰 액티비티를 프래그먼트로 전환
- [ ] [Result에 대해 알아보기](https://toss.tech/article/kotlin-result) Result를 이용해서 refreshToken이라는 함수가 제대로 완료되었을 경우 다음 로직 실행
- [ ] 검색 했다가 돋보기 버튼 누르면 카페 목록이 리사이클러뷰로 보이고 -> 카페 아이템 클릭해서 인포 페이지로 넘어갔다가 다시 뒤로가기 하면 검은 화면 보이는 점 변경하기
- [x] 화면 가장 위에 시간 등 표시되는 윈도우 활성화하기
- [x] 약관동의 체크박스: 모두 체크 했다가 하나만 체크 해제했는데 다 해제되는 것
- [x] 회원가입 페이지에 인풋텍스트필드가 많아짐에 따라 화면에 다 안 담기는 경우가 발생함. -> 리스트뷰로 전환하는 것 고려
- [x] 아직 미구현 서비스 눌렀을 때 "서비스 제공 예정입니다"와 같은 문구 ios쪽이랑 맞추기
- [x] 최근 본 매장에서 marginTop 조금 더 주기 (4dp 단위)
- [ ] ui 관련 코드들 presentation 모듈로 옮기기
- [x] 회원가입 시 인증번호 입력란 아이디 찾기와 동일하게 만들기
- [ ] 회원가입 Frag에서 ViewModel에서 수행해야 하는 기능들 옮기기
- [ ] 휴대폰 인증 할 때 입력한 번호로 인증 번호 받은 후에 번호 수정하면 다시 인증 코드 받으라는 메세지 띄우기
- [ ] 휴대폰 인증번호 발송 후 제한된 3분이 지나면 인증 못하게 막기(이 부분은 서버에서 처리가 될 수도 있음)
- [ ] 휴대폰 인증 시 '만료된 인증번호' 수정됐으니 수정하기

### 추천: 공부할 것
- [ ] Flow, StateFlow, SharedFlow

**Comment by YiBeomSeok** [comment]
@WooBinHam 
@Jaeeepp

**Reply by YiBeomSeok** [reply]
반응 달아주세용~

**Reply by WooBinHam** [reply]
확인했습니다!

**Comment by WooBinHam** [comment]
## 토큰관련 문제
1. 로그인 시 서버에게 accessToken과 refreshToken을 받음. 로그인 완료되면 마이페이지로 popBackStack. 이 상태에서 메인 페이지로 갔을 때 찜한 매장을 불러오는데 있어서 토큰이 제대로 안붙음. 앱을 재실행 해야 붙음. (토큰 재발급 후에도 마찬가지. 재실행을 해야 갱신된 토큰이 적용됨)
2. 토큰이 만료된 경우 CafeListFrag에서 찜한매장 불러오는데 오류가 있음. 그리고 로그인 하지 않은 경우에도 찜한매장을 요청함으로써 CafeListFrag에서 토스트 메시지가 뜸.

(추후 발견될 때마다 업뎃)

**Comment by WooBinHam** [comment]
1. 문자 인증번호 발송 시 휴대폰 번호 입력란 수정 막기
2. 카페 상세 페이지에서 위치 보기 누르면 MapFragment로 넘어가기
3. reviewId 나오면 리뷰 수정 삭제 구현하기
